code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -2,29 +2,135 @@
 	density = 1
 	anchored = 1
 	icon = 'icons/obj/cult.dmi'
+	var/cooldowntime = 0
 
+/obj/structure/cult/proc/getETA() //dunc and oranges are life-savers
+    var/time = round((0-(world.time-cooldowntime))/600, 1)
+    var/eta = "[time] minutes"
+    if(time == 1)
+        eta = "about one minute"
+    else if(time == 0)
+        eta = "less than a minute"
+    return eta
+         
 /obj/structure/cult/talisman
 	name = "altar"
 	desc = "A bloodstained altar dedicated to Nar-Sie"
 	icon_state = "talismanaltar"
-
+	
+/obj/structure/cult/talisman/attack_hand(mob/living/user)
+	if(!iscultist(user))
+		user << "<span class='warning'>You don't even begin to understand what these words mean...</span>"
+		return
+	if(cooldowntime)
+		user << "<span class='cultitalic'>The magic here is weak, it will be ready to use again in [getETA()]. </span>"
+		return
+	var/choice = alert(user,"You study the schematics etched into the forge...",,"Eldritch Whetstone","Zealot's Blindfold","Flask of Unholy Water")
+	switch(choice)
+		if("Eldritch Whetstone")
+			var/obj/item/weapon/sharpener/cult/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You kneel before the altar and your faith is rewarded with an [N.name]!</span>"
+		if("Zealot's Blindfold")
+			var/obj/item/clothing/glasses/night/cultblind/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You kneel before the altar and your faith is rewarded with a [N.name]!</span>"
+		if("Flask of Unholy Water")
+			var/obj/item/weapon/reagent_containers/food/drinks/bottle/unholywater/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You kneel before the altar and your faith is rewarded with a [N.name]!</span>"
+	cooldowntime = world.time + 2400
+	spawn(cooldowntime)
+	cooldowntime = 0
+	
 /obj/structure/cult/forge
 	name = "daemon forge"
 	desc = "A forge used in crafting the unholy weapons used by the armies of Nar-Sie"
 	icon_state = "forge"
+	luminosity = 3
+
+/obj/structure/cult/forge/attack_hand(mob/living/user)
+	if(!iscultist(user))
+		user << "<span class='warning'>You don't even begin to understand what these words mean...</span>"
+		return
+	if(cooldowntime)
+		user << "<span class='cultitalic'>The magic here is weak, it will be ready to use again in [getETA()]. </span>"
+		return
+	var/choice = alert(user,"You study the schematics etched into the forge...",,"Shielded Robe","Flagellant's Rags","Nar-Sian Hardsuit")
```

Comment:
This is a sleep() technically, so you gotta check for user and check for user in range.